lib/azure/armrest/insights/metrics_service.rb

```diff
@@ -64,7 +80,12 @@ def list_metrics(resource, filter = nil)
           )
 
           url << "?api-version=#{api_version}"
-          url << "&$filter=#{filter}" if filter
+
+          # The :filter option requires a leading '$'
+          options.each do |key, value|
+            key.to_s == 'filter' ? url << "&$" : url << "&"
```

Comment:
Does it maintain the backward compatibility? The old code directly accepts the value of the filter without key `filter`.

Reply:
@bzwei It doesn't. I guess I *could* keep backwards compatibility for a time and check to see if `options` is a string instead of a hash.

Reply:
@bzwei Ok, updated.